### PR TITLE
Add wildcard cache key deletion for device type cache purge

### PR DESCRIPTION
**Pull request by pradeep910**
Adds wildcard check to delete device type cache such as `<URL>--*`.

This requires following filter as URLs will not contain `--*` by default.

```php
add_filter( 'rt_nginx_helper_purge_url', function( $url ) {
       // Add wildcard in cache key URL for all device type keys deletion.
       $url = $url . '--*';
       return $url;
});
```

For #202

### CODE REVIEW

File: admin/class-phpredis-purger.php [hunk 1]
```diff
@@ -106,7 +106,7 @@ public function purge_url( $url, $feed = true ) {
 		/**
 		 * Filters the URL to be purged.
 		 *
-		 * @since 2.1.0
+		 * @since 2.0.4
```
Comment (chandrapatel) [inline admin/class-phpredis-purger.php]: @pradeep910 Introducing of filters/actions is major change so I'd suggest changing plugin version to `2.1.0`.

File: admin/class-phpredis-purger.php [hunk 2]
```diff
@@ -120,12 +120,44 @@ public function purge_url( $url, $feed = true ) {
...
+		 * Regardless of what key / suffix is being to store `$device_type` cache , it will be deleted.
+		 *
+		 * @since 2.0.4
+		 */
+		if ( strpos( $_url_purge_base, '*' ) === false ) {
+
+			$status = $this->delete_single_key( $_url_purge_base );
+
+			if ( $status ) {
+				$this->log( '- Purge URL | ' . $_url_purge_base );
```
Comment (chandrapatel) [inline admin/class-phpredis-purger.php]: `$this->log( '- Purged URL | ' . $_url_purge_base . ' | ' . $status );`

File: admin/class-phpredis-purger.php [hunk 3]
```diff
@@ -120,12 +120,44 @@ public function purge_url( $url, $feed = true ) {
...
+				$this->log( '- Cache Not Found | ' . $_url_purge_base, 'ERROR' );
+			}
+
 		} else {
-			$this->log( '- Cache Not Found | ' . $url, 'ERROR' );
+
+			$status = $this->delete_keys_by_wildcard( $_url_purge_base );
+
+			if ( $status ) {
+				$this->log( '- Purge Wild Card URL | ' . $_url_purge_base . ' | ' . $status . ' url purged' );
```
Comment (chandrapatel) [inline admin/class-phpredis-purger.php]: `$this->log( '- Purged Wild Card URL | ' . $_url_purge_base . ' | ' . $status . ' url purged' );`

File: admin/class-phpredis-purger.php [hunk 4]
```diff
@@ -166,9 +198,9 @@ public function custom_purge_urls() {
 					$status    = $this->delete_single_key( $purge_url );
 
 					if ( $status ) {
-						$this->log( '- Purge URL | ' . $purge_url );
+						$this->log( '- [phpredis] Purge URL | ' . $purge_url );
 					} else {
-						$this->log( '- Cache Not Found | ' . $purge_url, 'ERROR' );
+						$this->log( '- [phpredis] Cache Not Found | ' . $purge_url, 'ERROR' );
```
Comment (chandrapatel) [inline admin/class-phpredis-purger.php]: Remove `[phpredis]` from log message. Its not required.

File: admin/class-predis-purger.php [hunk 5]
```diff
@@ -119,7 +119,45 @@ public function purge_url( $url, $feed = true ) {
...
+		 * @since 2.0.4
+		 */
+		if ( strpos( $_url_purge_base, '*' ) === false ) {
+
+			$status = $this->delete_single_key( $_url_purge_base );
+
+			if ( $status ) {
+				$this->log( '- [predis] Purge URL | ' . $_url_purge_base );
+			} else {
+				$this->log( '- [predis] Cache Not Found | ' . $_url_purge_base, 'ERROR' );
```
Comment (chandrapatel) [inline admin/class-predis-purger.php]: Remove `[predis]` from log message. Its not required.

File: admin/class-phpredis-purger.php (hunk 3 above)
Comment (pradeep910) [reply admin/class-phpredis-purger.php]: This thing we will have to do everywhere. I think keeping `Purge` is fine as it's an action. 
On the same line we see `' . $status . ' url purged'`.

File: admin/class-phpredis-purger.php (hunk 1 above)
Comment (pradeep910) [reply admin/class-phpredis-purger.php]: Ok changing to 2.1.0

**Comment by radhe** [comment]
@kirtangajjar @th3karkota 
can you do the testing?

**Comment by pradeep910** [comment]
We've merged this plugin after testing it on one of our live sites.
cc @chandrapatel @kirtangajjar @radhe

**Pull request merged**